_studio/mfx_lib/shared/src/libmfxsw_enc.cpp

```diff
@@ -46,8 +46,10 @@
 template<>
 VideoENC* _mfxSession::Create<VideoENC>(mfxVideoParam& par)
 {
-    VideoENC* pENC = nullptr;
+#if ( defined(MFX_ENABLE_LA_H264_VIDEO_HW) || defined(MFX_ENABLE_H264_VIDEO_FEI_PREENC) || defined(MFX_ENABLE_H264_VIDEO_FEI_ENC))
```

Comment:
I suggest to directly use m_pCore.get() below, i.e.:
pPAK = (VideoENC*) new VideoENC_ENC(m_pCore.get(), &mfxRes);
Less macros the better...